### PR TITLE
RHIROS-518 added modal to manage systems table columns

### PR DESCRIPTION
**Pull request by PreetiW**
### Description

* Added modal to control the visible columns
* Added new column `operating system`

**Jira:** https://issues.redhat.com/browse/RHIROS-518

**Reference Links**

* [Patternfly Modal](https://www.patternfly.org/v4/components/modal/)
* [Data list](https://www.patternfly.org/v4/components/data-list/#datalist)
* [SplitItem](https://www.patternfly.org/v4/layouts/split/#splititem)
* [Inventory Columns](https://github.com/RedHatInsights/insights-inventory-frontend/blob/a4a0c054c9eb84e318a76422cac00df52781f0b5/doc/props_table.md#columns)
* [Redux](https://github.com/RedHatInsights/frontend-components/blob/4c5456bd2b27497fbc8849bf5aad99be1d6fb755/packages/utils/doc/redux.md#redux)
* [ActionsConfig](https://github.com/RedHatInsights/frontend-components/blob/4c5456bd2b27497fbc8849bf5aad99be1d6fb755/packages/components/doc/primary_toolbar.md#primary-toolbar)

**Demo link:**

https://drive.google.com/file/d/1za7tB2XV0UfxW2VeZRedeI7dDF5Ii8Lk/view?usp=sharing

**Screenshot**

image

**Relevant Backend PR:**
https://github.com/RedHatInsights/ros-backend/pull/164

### CODE REVIEW

File: src/Components/RosTable/RenderColumn.js [hunk 1]
```diff
@@ -46,3 +38,11 @@ export const DiskUsageData = (data, id, item) => {
             </Tooltip>
     );
 };
+
+export const displayOS = (data, id, { state }) => {
+    return (
+        state === NO_DATA_STATE ?
+            <span>{ NO_DATA_VALUE }</span> :
+            <span>{ data }</span>
```
Comment (kgaikwad) [inline src/Components/RosTable/RenderColumn.js:46]: This is not the case, you can directly display OS info. On backend, it is getting OS info from inventory topic so this information will be there regarding less of rule hit information.
You can handle condition if that value is none.

**Review by kgaikwad — CHANGES_REQUESTED** [review]
Overall looks good! Added small inline comment.

File: src/Components/RosTable/RenderColumn.js (hunk 1 above)
Comment (PreetiW) [reply src/Components/RosTable/RenderColumn.js]: good catch @kgaikwad I updated the condition and checked it against the backend. Thanks!

File: src/Routes/RosPage/RosPage.js [hunk 2]
```diff
@@ -198,14 +194,32 @@ class RosPage extends React.Component {
...
 
         return (
             this.props.showConfigSteps
                 ?   <ServiceNotConfigured/>
                 :   <Card className='pf-t-light  pf-m-opaque-100'>
                     <CardBody>
+                        <ManageColumnsModal
+                            isModalOpen={this.state.isColumnModalOpen}
+                            setModalOpen={this.setColumnModalOpen}
+                            activeColumns={this.props.columns}
```
Comment (r14chandra) [inline src/Routes/RosPage/RosPage.js:220]: @PreetiW Could you rename this prop to something else (as discussed) so as to avoid confusion.

**Review by r14chandra — CHANGES_REQUESTED** [review]
Everything looks great! 🚀 Just left a comment above.

File: src/Routes/RosPage/RosPage.js (hunk 2 above)
Comment (PreetiW) [reply src/Routes/RosPage/RosPage.js]: Thanks @r14chandra for the review, I have refactored the prop name and have pushed the changes.

**Review by r14chandra — APPROVED** [review]
Tested 🚀 ! It's working as expected 👍🏻  @PreetiW

**Review by kgaikwad — APPROVED** [review]
Looks good to me!

**Pull request merged**